python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -2990,6 +2991,74 @@ def get_latest_planned_materialization_info(
             run_id=records[0].run_id,
         )
 
+    def get_updated_data_version_partitions(
+        self, asset_key: AssetKey, partitions: Iterable[str], since_storage_id: int
+    ) -> Set[str]:
+        latest_data_version_event_ids_subquery = self._latest_event_ids_by_partition_subquery(
```

Comment:
If I'm reading this correctly, doesn't this return the wrong value in the case that:

1. Asset A is materialized with data version 1
2. Asset A is materialized with data version 2
3. Asset A is materialized with no data version

? the latest data version event ids subquery would give the storage id of (3), which would mean that it would not have a match when joined on the AssetEventTagsTable.

Similarly, I think the following sequence of events would have an issue:

1. A materialized with data version 1
2. A materialized with data version 2
3. A materialized with data version 2

In this case, the latest data version storage id would link to event (3), and the penultimate data version storage id would link to event (2). This means that we would always return that A had not been updated (the data versions are identical) regardless of the `since_storage_id` that's passed in. But in reality, if a since_storage_id that's less than event (2) is passed in, then we should consider A to be updated since that cursor, even if the most recent event did not do that update

Seems like these queries need to factor in the asset event tags -- maybe we can skip the `_latest_event_ids_by_partition_subquery` entirely and just do queries against `AssetEventTags`? That'd help filter out `None` data version tags for the first bit, and then we could query for data version tags that != the latest non-`None` data version tag for the second part.

Reply:
I can change the implementation if we think it's a problem, but I'm pretty sure that this will be handled by the join.  I've added a test that I think illustrates that the implementation works with your second scenario.

Reply:
you were right... updated to query the joined tags/event log tables.  We can't query asset tags directly since we don't pull out the partition tag directly.